node/src/block_producer/block_producer_reducer.rs

```diff
@@ -304,8 +304,24 @@ impl BlockProducerEnabled {
                     bug_condition!("Invalid state for `BlockProducerAction::BlockProduced` expected: `BlockProducerCurrentState::BlockProveSuccess`, found: {:?}", current_state);
                 }
 
+                #[cfg(feature = "p2p-libp2p")]
+                let to_broadcast =
+                    if let BlockProducerCurrentState::Produced { block, .. } = &state.current {
+                        Some(v2::MinaBlockBlockStableV2::clone(&block.block))
+                    } else {
+                        None
+                    };
+
                 let dispatcher = state_context.into_dispatcher();
                 dispatcher.push(BlockProducerAction::BlockInject);
+
+                #[cfg(feature = "p2p-libp2p")]
+                if let Some(block) = to_broadcast {
+                    use mina_p2p_messages::gossip::GossipNetMessageV2;
+
+                    let message = Box::new(GossipNetMessageV2::NewState(block));
+                    dispatcher.push(p2p::P2pNetworkPubsubAction::Broadcast { message });
```

Comment:
why is it not broadcasted after the `Inject` above? this means that it will be broadcasted even if the transition frontier rejects it right?

Reply:
(I don't mean inside the `BlockInject` action itself but as the result of it, same way as other broadcasted blocks, after successful application and integration into the frontier)